src/plugins/intel_cpu/src/transformations/snippets/tpp/op/brgemm.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+#include "modifiers.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace tpp {
+namespace op {
+
+/**
+ * @interface BrgemmTPP
+ * @brief BrgemmTPP is a batch-reduced matrix multiplication with the support of arbitrary strides between matrices rows
+ *        with support of several precisions on plugin level
+ * @ingroup snippets
+ */
+class BrgemmTPP : virtual public modifier::TensorProcessingPrimitive, public snippets::op::Brgemm {
```

Comment:
I'd suggest to remove `tpp` from name since the class in `tpp` namespace. C++ specially provides namespace to distinguish classes with the same names but different implementation. What do you think? It's just a suggestion to discuss :)
Maybe we should rename `Brgemm` to `BrgemmCPU` as well)))

Reply:
Exactly, this naming was specifically introduced to align with the `BrgemmCPU` naming.
Let's discuss it on the sync